packages/nimble-components/src/wafer-map/modules/experimental/hover-handler.ts

```diff
@@ -0,0 +1,79 @@
+import type { WaferMap } from '../..';
+import { PointCoordinates, WaferMapOriginLocation } from '../../types';
+
+/**
+ * HoverHandler deals with user interactions and events like hovering
+ */
+export class HoverHandler {
+    public constructor(private readonly wafermap: WaferMap) {}
+
+    public mousemove(event: MouseEvent): void {
+        if (this.wafermap.diesTable === undefined) {
+            this.wafermap.hoverDie = undefined;
+            return;
+        }
+        // get original mouse position in case we are in zoom.
+        const invertedPoint = this.wafermap.transform.invert([
+            event.offsetX,
+            event.offsetY
+        ]);
+
+        // does not work yet until data manager will parse diesTable
+        const dieCoordinates = this.calculateDieCoordinates(this.wafermap, {
+            x: invertedPoint[0],
+            y: invertedPoint[1]
+        });
+        const colIndex = this.wafermap.diesTable
+            .getChild('colIndex')!
+            .toArray() as Int32Array;
+        const rowIndex = this.wafermap.diesTable
+            .getChild('rowIndex')!
+            .toArray() as Int32Array;
+
+        for (let i = 0; i < colIndex.length; i++) {
+            if (
+                colIndex[i] === dieCoordinates.x
+                && rowIndex[i] === dieCoordinates.y
+            ) {
+                this.wafermap.hoverDie = {
+                    index: i,
+                    x: dieCoordinates.x,
+                    y: dieCoordinates.y
+                };
+                return;
+            }
+        }
+        this.wafermap.hoverDie = undefined;
+    }
+
+    public mouseout(): void {
+        this.wafermap.hoverDie = undefined;
+    }
+
+    private calculateDieCoordinates(
```

Comment:
This almost seems like a generally useful function that could be needed for some other function beyond hovering. Do you think it makes sense to pull this out into a static utility class?